src/quick-lint-js/parse.h

```diff
@@ -2460,6 +2460,71 @@ class parser {
     v.visit_exit_with_scope();
   }
 
+  bool is_side_effect_free(expression *ast) {
+    switch (ast->kind()) {
+    case expression_kind::_class:
```

Comment:
Blocking: Class expressions might have side effects. For example:
```js
function f() { console.log("yo"); }
let C = class { [f()]() {} };
```

We don't have enough information in the AST to know, so let's be conservative and assume that side effects might happen.